cmd/gitops-server/main.go

```diff
@@ -52,7 +55,34 @@ func NewAPIServerCommand() *cobra.Command {
 				return fmt.Errorf("could not create kube http client: %w", err)
 			}
 
-			profilesConfig := server.NewProfilesConfig(rawClient, "default", "weaveworks-charts")
+			tmpDir, err := os.MkdirTemp("", "helmCacheDir")
+			if err != nil {
+				return fmt.Errorf("failed to create helm cache: %w", err)
+			}
+
+			helmCache, err := cache.NewCache(tmpDir)
+			if err != nil {
+				return fmt.Errorf("failed to create helm cache: %w", err)
+			}
+
+			helmWatcher, err := watcher.NewWatcher(rawClient, helmCache)
+			if err != nil {
+				return fmt.Errorf("failed to create watcher: %w", err)
+			}
+
+			go func() {
+				if err := helmWatcher.StartWatcher(); err != nil {
+					appConfig.Logger.Error(err, "failed to start the watcher")
+					os.Exit(1)
+				}
+			}()
+
+			// Create the cache here as well and pass it in through the profiles Config thing.
+			runtimeNamespace := os.Getenv("RUNTIME_NAMESPACE")
```

Comment:
This is less than ideal. But I'm not sure how to pass in anything here.

Reply:
How about looking at how this works https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/blob/master/main.go#L86-L90 ?

Reply:
The namespace a controller watches is not the same thing as the runtime namespace (the ns where it's been deployed). The runtime namespace is mapped from the Kubernetes Downward API.

Reply:
That is perfectly true. I added this as a workaround. This definitely is not right and I'm not sure what to do here about it. We had some hardcoded things in here which said, `default` and `weaveworks-charts` or something like that. But that's definitely not true and not even desired for this to work. So I'm trying to find a better way to do this.

Reply:
In Flux source-controller, we use the runtime namespace value, then we advertise it for other controllers in the status sub-resource. This way all the other controllers can determine at runtime where the source cache is located.

Reply:
For example, the thing I'm testing has a helmchart for profile in wego-system with name `podinfo`. And even so I can only use that when I configure the namespace from the command line when I'm running `gitops ui run --helm-repo-namespace wego-system --helm-repo-name podinfo`. So right now, my hands I tied in this. This is how this is designed for the moment, and later we can improve upon that. For now, I guess only a single HelmRepository will be a profile.